### PR TITLE
Scala 2.13.0 rc1

### PR DESCRIPTION
**Pull request by pjfanning**
A rebased version of https://github.com/slick/slick/pull/1966 - there is still work to do to check if everything from 1966 is included.

### CODE REVIEW

**Comment by hvesalai** [comment]
At least it compiles :-) 

A simple `diff` of the two branches should tell you if you have everything or not.

**Comment by SethTisue** [comment]
fantastic! wonderful to see this

**Comment by SethTisue** [comment]
community build: `[info] Project slick-----------------------: SUCCESS (unchanged, not rebuilt)` 🎉

File: build.sbt [hunk 1]
```diff
@@ -8,7 +8,9 @@ version in ThisBuild := "3.4.0-SNAPSHOT"
 // The next release to be cut from master will be 3.4.0 during develop of 3.4.0 we check compatibility with 3.3.0.
 // The goal is not to stop any breaking change, but to make us aware. For each breaking change we should add MiMa exclusions.
 // This will also help us decide when a PR can be backported in 3.3.x branch.  
-binaryCompatSlickVersion in ThisBuild := Some("3.3.0")
```
Comment (nafg) [inline build.sbt:11]: Why was this necessary?

Comment (hvesalai) [reply build.sbt:11]: Yeah, the comment tried to explain how to deal with MIMA warnings and this was not what was intended. The idea was that we could communicate the breaking changes if we saw all the excludes.
